build-homepage script fails on Debian 9

Running the script `scripts/build-homepage` fails during cmake compiler testing phase.

```
root@elektrahomepage:/# lsb_release -a
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 9.1 (stretch)
Release:	9.1
Codename:	stretch
```
- Elektra Version: master

# Further Log Files and Output

```
+ cmake -DENABLE_ASAN=ON -DBUILD_FULL=OFF -DBUILD_SHARED=ON -DBUILD_STATIC=OFF -DBUILD_DOCUMENTATION=OFF -DPLUGINS=ALL;-fstab;-semlock;file;xerces -DTOOLS=kdb;rest-backend;rest-frontend -DCMAKE_C_FLAGS=-D_FORTIFY_SOURCE=2 -Werror=format-security -fstack-protector-strong -Wstack-protector -fPIE -pie -DCMAKE_CXX_FLAGS=-D_FORTIFY_SOURCE=2 -Werror=format-security -fstack-protector-strong -Wstack-protector -fPIE -pie -DCMAKE_EXE_LINKER_FLAGS=-Wl,-z,now -Wl,-z,relro ..
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- broken
CMake Error at /usr/share/cmake-3.7/Modules/CMakeTestCCompiler.cmake:51 (message):
  The C compiler "/usr/bin/cc" is not able to compile a simple test program.
...
  Building C object CMakeFiles/cmTC_56393.dir/testCCompiler.c.o
  /usr/bin/cc -D_FORTIFY_SOURCE=2 -Werror=format-security
  -fstack-protector-strong -Wstack-protector -fPIE -pie -o
  CMakeFiles/cmTC_56393.dir/testCCompiler.c.o -c
  /home/jenkins/workspace/workspace/elektra-homepage/build/CMakeFiles/CMakeTmp/testCCompiler.c
  cc1: error: -Wformat-security ignored without -Wformat
  [-Werror=format-security]
  cc1: some warnings being treated as errors
  CMakeFiles/cmTC_56393.dir/build.make:65: recipe for target
  'CMakeFiles/cmTC_56393.dir/testCCompiler.c.o' failed
  make[1]: *** [CMakeFiles/cmTC_56393.dir/testCCompiler.c.o] Error 1

--- a/scripts/build-homepage
+++ b/scripts/build-homepage
@@ -2,13 +2,27 @@
 # This script is sourced on build server, shebang is for convenience
 # bash is used to have a predictable ulimit behavior
 
+set -e
+set -x
+
 # remember the build directory
 ROOT_DIR=$PWD
 
 # build the applications
 mkdir -p build
 cd build
-cmake -DENABLE_ASAN=ON -DBUILD_FULL=OFF -DBUILD_SHARED=ON -DBUILD_STATIC=OFF -DBUILD_DOCUMENTATION=OFF -DPLUGINS='ALL;-fstab;-semlock;file;xerces' -DTOOLS='kdb;rest-backend;rest-frontend' -DCMAKE_C_FLAGS='-D_FORTIFY_SOURCE=2 -Werror=format-security -fstack-protector-strong -Wstack-protector -fPIE -pie' -DCMAKE_CXX_FLAGS='-D_FORTIFY_SOURCE=2 -Werror=format-security -fstack-protector-strong -Wstack-protector -fPIE -pie' -DCMAKE_EXE_LINKER_FLAGS='-Wl,-z,now -Wl,-z,relro' ..
+
+C_FLAGS='-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-protector-strong -Wstack-protector -fPIE -pie'
+LD_FLAGS='-Wl,-z,now -Wl,-z,relro'
+
+cmake -DENABLE_ASAN=ON -DBUILD_FULL=OFF -DBUILD_SHARED=ON -DBUILD_STATIC=OFF -DBUILD_DOCUMENTATION=OFF \
+        -DPLUGINS='ALL;-fstab;-semlock;file;xerces' \
+        -DTOOLS='kdb;rest-backend;rest-frontend' \
+        -DCMAKE_C_FLAGS="$C_FLAGS" \
+        -DCMAKE_CXX_FLAGS="$C_FLAGS" \
+        -DCMAKE_EXE_LINKER_FLAGS="$LD_FLAGS" \
+        ..
+
 make -j 3
 
 # test the applications (TODO: external fails?)

--- a/scripts/build-homepage
+++ b/scripts/build-homepage
@@ -25,9 +25,8 @@ cmake -DENABLE_ASAN=ON -DBUILD_FULL=OFF -DBUILD_SHARED=ON -DBUILD_STATIC=OFF -DB
 
 make -j 3
 
-# test the applications (TODO: external fails?)
-#make run_all
-ctest --output-on-failure -E external
+# run tests (exclude tests which write in system paths)
+make run_nokdbtests
 
 # rotate backup of previous website
 rm -rf /usr/local/share/elektra/tool_data_backup

--- a/scripts/build-homepage
+++ b/scripts/build-homepage
@@ -60,9 +60,11 @@ ulimit -c unlimited
 # place for core dumps (name by: echo "core_%t_%p" > /proc/sys/kernel/core_pattern)
 cd ${INSTALL_PATH}
 
+PID_FILE=`kdb sget /sw/elektra/restbackend/#0/current/cppcms/daemon/lock /run/elektra-rest-backend.pid`
+
 ps -ef
 printf "Old PID was: "
-cat /run/elektra-rest-backend.pid || /bin/true
+cat ${PID_FILE} || /bin/true
 echo
 
 
@@ -78,7 +80,7 @@ netstat -tlpen
 kdb stop-rest-backend || /bin/true
 
 # cleanup /tmp files from build
-find /tmp -mindepth 1 -delete
+find /tmp -mindepth 1 -user `id -u` -delete || true
 
 # avoid 'address already in use'
 while netstat -tlpen | grep "$IP:$PORT"
@@ -90,7 +92,7 @@ done
 kdb run-rest-backend
 
 printf "New PID is: "
-cat /run/elektra-rest-backend.pid || /bin/true
+cat ${PID_FILE} || /bin/true
 echo
 
 netstat -tlpen
@@ -102,12 +104,15 @@ netstat -tlpen
 #   required files afterwards, ensuring an always up-to-date deployment
 kdb build-rest-frontend
 
-# create and copy api description
 ELEKTRA_VERSION=$(kdb get --no-newline system/elektra/version/constants/KDB_VERSION)
-apiary preview --path=$ROOT_DIR/doc/api_blueprints/snippet-sharing.apib --output=/srv/libelektra/restapi/$ELEKTRA_VERSION/snippet-sharing.html
-cp $ROOT_DIR/doc/api_blueprints/snippet-sharing.apib /srv/libelektra/restapi/$ELEKTRA_VERSION/snippet-sharing.apib
-rm /srv/libelektra/restapi/current
-ln -s /srv/libelektra/restapi/$ELEKTRA_VERSION /srv/libelektra/restapi/current
+API_DOC_DIR="${ROOT_DIR}/build/API_DOC/restapi/${ELEKTRA_VERSION}"
+
+mkdir -p ${API_DOC_DIR}
+
+# create and copy api description
+apiary preview --path=$ROOT_DIR/doc/api_blueprints/snippet-sharing.apib --output=${API_DOC_DIR}/snippet-sharing.html
+cp $ROOT_DIR/doc/api_blueprints/snippet-sharing.apib ${API_DOC_DIR}/snippet-sharing.apib
+(cd ${API_DOC_DIR}/.. ; rm -f current ; ln -s $ELEKTRA_VERSION current )
 
 ps -ef
 ulimit -a

--- a/scripts/build-homepage
+++ b/scripts/build-homepage
@@ -12,10 +12,13 @@ ROOT_DIR=$PWD
 mkdir -p build
 cd build
 
+INSTALL_PATH="/usr/local/elektra_backend"
+
 C_FLAGS='-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-protector-strong -Wstack-protector -fPIE -pie'
 LD_FLAGS='-Wl,-z,now -Wl,-z,relro'
 
 cmake -DENABLE_ASAN=ON -DBUILD_FULL=OFF -DBUILD_SHARED=ON -DBUILD_STATIC=OFF -DBUILD_DOCUMENTATION=OFF \
+	-DCMAKE_INSTALL_PREFIX="${INSTALL_PATH}"  -DINSTALL_SYSTEM_FILES=OFF \
         -DPLUGINS='ALL;-fstab;-semlock;file;-xerces' \
         -DTOOLS='kdb;rest-backend;rest-frontend' \
         -DCMAKE_C_FLAGS="$C_FLAGS" \
@@ -35,8 +38,14 @@ cp -ra /usr/local/share/elektra/tool_data /usr/local/share/elektra/tool_data_bac
 # if tests were ok, we can install
 make install
 
+
+# use installed binaries and libs
+export PATH=${INSTALL_PATH}/bin:${PATH}
+export LD_LIBRARY_PATH=${INSTALL_PATH}/lib
+
 # now lets first output some version information for the build log
 kdb --version
+
 # print key database for debugging
 #kdb export / simpleini
 

--- a/scripts/build-homepage
+++ b/scripts/build-homepage
@@ -9,22 +9,24 @@ set -x
 ROOT_DIR=$PWD
 
 # build the applications
-mkdir -p build
-cd build
+rm -rf build && mkdir -p build && cd build
 
-INSTALL_PATH="/usr/local/elektra_backend"
+# concrete install location is configurable
+if [ -z "${INSTALL_PATH}" ] ; then
+	INSTALL_PATH="/usr/local/elektra_backend"
+fi
 
 C_FLAGS='-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-protector-strong -Wstack-protector -fPIE -pie'
 LD_FLAGS='-Wl,-z,now -Wl,-z,relro'
 
 cmake -DENABLE_ASAN=ON -DBUILD_FULL=OFF -DBUILD_SHARED=ON -DBUILD_STATIC=OFF -DBUILD_DOCUMENTATION=OFF \
 	-DCMAKE_INSTALL_PREFIX="${INSTALL_PATH}"  -DINSTALL_SYSTEM_FILES=OFF \
-        -DPLUGINS='ALL;-fstab;-semlock;file;-xerces' \
-        -DTOOLS='kdb;rest-backend;rest-frontend' \
-        -DCMAKE_C_FLAGS="$C_FLAGS" \
-        -DCMAKE_CXX_FLAGS="$C_FLAGS" \
-        -DCMAKE_EXE_LINKER_FLAGS="$LD_FLAGS" \
-        ..
+	-DPLUGINS='ALL;-fstab;-semlock;file;-xerces;-ruby;-lua;-python;-python2' \
+	-DTOOLS='kdb;rest-backend;rest-frontend' \
+	-DCMAKE_C_FLAGS="$C_FLAGS" \
+	-DCMAKE_CXX_FLAGS="$C_FLAGS" \
+	-DCMAKE_EXE_LINKER_FLAGS="$LD_FLAGS" \
+	..
 
 make -j 3
 
@@ -32,8 +34,8 @@ make -j 3
 make run_nokdbtests
 
 # rotate backup of previous website
-rm -rf /usr/local/share/elektra/tool_data_backup
-cp -ra /usr/local/share/elektra/tool_data /usr/local/share/elektra/tool_data_backup
+rm -rf ${INSTALL_PATH}/share/elektra/tool_data_backup
+cp -ra ${INSTALL_PATH}/share/elektra/tool_data ${INSTALL_PATH}/share/elektra/tool_data_backup
 
 # if tests were ok, we can install
 make install

--- a/scripts/build-homepage
+++ b/scripts/build-homepage
@@ -16,7 +16,7 @@ C_FLAGS='-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-protector-
 LD_FLAGS='-Wl,-z,now -Wl,-z,relro'
 
 cmake -DENABLE_ASAN=ON -DBUILD_FULL=OFF -DBUILD_SHARED=ON -DBUILD_STATIC=OFF -DBUILD_DOCUMENTATION=OFF \
-        -DPLUGINS='ALL;-fstab;-semlock;file;xerces' \
+        -DPLUGINS='ALL;-fstab;-semlock;file;-xerces' \
         -DTOOLS='kdb;rest-backend;rest-frontend' \
         -DCMAKE_C_FLAGS="$C_FLAGS" \
         -DCMAKE_CXX_FLAGS="$C_FLAGS" \

--- a/scripts/build-homepage
+++ b/scripts/build-homepage
@@ -43,7 +43,7 @@ make install
 
 # use installed binaries and libs
 export PATH=${INSTALL_PATH}/bin:${PATH}
-export LD_LIBRARY_PATH=${INSTALL_PATH}/lib
+export LD_LIBRARY_PATH=${INSTALL_PATH}/lib:${INSTALL_PATH}/lib/elektra
 
 # now lets first output some version information for the build log
 kdb --version
@@ -58,9 +58,9 @@ BUILD_ID=dontKillMe
 ulimit -c unlimited
 
 # place for core dumps (name by: echo "core_%t_%p" > /proc/sys/kernel/core_pattern)
-cd /var/lib/jenkins
+cd ${INSTALL_PATH}
 
-pstree
+ps -ef
 printf "Old PID was: "
 cat /run/elektra-rest-backend.pid || /bin/true
 echo
@@ -109,5 +109,5 @@ cp $ROOT_DIR/doc/api_blueprints/snippet-sharing.apib /srv/libelektra/restapi/$EL
 rm /srv/libelektra/restapi/current
 ln -s /srv/libelektra/restapi/$ELEKTRA_VERSION /srv/libelektra/restapi/current
 
-pstree
+ps -ef
 ulimit -a

